### ISSUE TITLE
Using a custom connection to wrap a specific field

### ISSUE THREAD

**Issue by anujbiyani** [rmosolgo/graphql-ruby#2621]
I'm trying to implement my own connection for a field. (I _think_ I need a custom connection because I want to paginate off of `Measurement#collected_at` instead of the ID, and I couldn't see how to do that from the docs without building my own connection.)

I've read through the [custom connections documentation](https://graphql-ruby.org/relay/connections.html#custom-connections) and got most of this working, with one issue: when I register the connection I have to target `ActiveRecord::AssociationRelation` because that's what my field is returning:
```ruby
GraphQL::Relay::BaseConnection.register_connection_implementation(ActiveRecord::AssociationRelation, Connections::MeasurementConnection)
```

This would, however, break any other field that wants to use the default connection implementation.

**Is there a way to use my custom connection class for only certain fields?**

I can use this workaround, but it's pretty ugly:
```ruby
# type file
class MeasurementRelationWrapper < ActiveRecord::AssociationRelation
end

class Types::VitalsType < Types::BaseObject
  field :heart_rate, Types::MeasurementType.connection_type, null: false
  def heart_rate
    measurements = object.some_association_that_gets_measurements
    klass = measurements.instance_variable_get(:@klass)
    association = measurements.instance_variable_get(:@association)

    MeasurementRelationWrapper.new(klass, association)
  end
end

# schema file
GraphQL::Relay::BaseConnection.register_connection_implementation(MeasurementRelationWrapper, Connections::MeasurementConnection)
```

**Comment by rmosolgo** [comment]
Hey! It sounds like you want to return a relation ordered by `collected_at`, and expose it as a connection, is that right? 

It _should_ just work, out of the box. In fact, the built-in implementation here uses the _index_ as a cursor, so it doesn't care how the relation is ordered. 

I expect this to work: 

```ruby 
  field :heart_rate, Types::MeasurementType.connection_type, null: false

  def heart_rate
    measurements = object.some_association_that_gets_measurements
    measurements.order(:collected_at) 
  end
```

Under the hood, GraphQL-Ruby will generate cursors using the returned ActiveRecord::Relation. 

Want to give it a try? Let me know how it goes!

**Comment by anujbiyani** [comment]
Ah right...that makes sense, and it does work. Thanks!

Is it possible to change the column your cursor is based on? In my custom connection I was also implementing stable cursors calculated from `collected_at` instead of `id` (and so also use `collected_at` in the `where` clause). I guess stable pagination is Pro feature so maybe this is possible with the Pro version?

My thinking is that if I'm ordering by `collected_at` but the `where` clause for the cursor is on `id`, I can still get unstable pagination.

**Comment by rmosolgo** [comment]
> possible with the Pro version?

Yes, it would work out-of-the-box with Pro's stable connections. That connection implementation _detects_ the ordering of the relation, then uses that ordering to add some `SELECT`s and `WHERE` clauses. 

But you can also implement it yourself -- in that case, you'd need to implement a connection implementation. In fact, `1.10.0.pre1` has a new pagination system which is much more flexible. For example, it supports one-off connection wrappers, so that some AR::Relations get default wrapper, but your special case could get a custom wrapper. 

You can find the docs for that future version in the source here: https://github.com/rmosolgo/graphql-ruby/tree/1.10-dev/guides/pagination. 

Please let me know how it goes if you decide to give it a try! There's also some early feedback about those docs (which I haven't addressed yet 😅 ) at #2542, in case that helps to make sense of it.

**Comment by anujbiyani** [comment]
Ooh excellent, I can remove my hack in 1.10 then. Thanks so much for your responses and all your hard work on this gem!
